File: Problemset/0806-Number of Lines To Write String/Simulation.py
class Solution:
    def numberOfLines(self, widths: List[int], S: str) -> List[int]:
        ret = [1, 0]
        for ch in S:
            if ret[1] + widths[ord(ch) - 97] > 100:
                ret[0] += 1
                ret[1] = 0
            ret[1] += widths[ord(ch) - 97]

        return ret

File: Problemset/0059-Spiral Matrix II/Simulation.rs
impl Solution {
    pub fn generate_matrix(n: i32) -> Vec<Vec<i32>> {
        let n = n as usize;
        let mut ret = vec![vec![0; n]; n];

        let mut num = 1;
        for i in 0..((n + 1) / 2) {
            for c in i..(n - 1 - i) {
                ret[i][c] = num;
                num += 1;
            }
            for r in i..(n - 1 - i) {
                ret[r][n - 1 - i] = num;
                num += 1;
            }
            for c in ((i + 1)..(n - i)).rev() {
                ret[n - 1 - i][c] = num;
                num += 1;
            }
            for r in ((i + 1)..(n - i)).rev() {
                ret[r][i] = num;
                num += 1;
            }
        }
        ret[n / 2][(n - 1) / 2] = (n * n) as i32;

        ret
    }
}
